src/Skipper/Matcher/FileInfoMatcher.php

```diff
@@ -61,6 +61,11 @@ private function doesFileMatchPattern(string $filePath, string $ignoredPath): bo
             return true;
         }
 
+        // realpathMatcher cannot resolve wildcards -> return early to prevent unnecessary IO
+        if ($ignoredPath[0] === '*' && $ignoredPath[strlen($ignoredPath) - 1] === '*') {
```

Comment:
Personally, I don't think this optimization is really needed, as while improve performance, we don't cover changed behaviour, normal `fnmatch` then `realpath` should be **safe** to cover both side.

--------

But if it really needed, `filePath` last char seems need to be checked just to be sure no relative path named: `foo.php*`

```suggestion
        if ($ignoredPath[0] === '*' && $ignoredPath[strlen($ignoredPath) - 1] === '*' && ! str_ends_with($filePath, '*') {
```

also, I think we need more tests on https://github.com/rectorphp/rector-src/blob/main/tests/Skipper/Skipper/SkipperTest.php to avoid merge and revert:

![Screenshot 2024-04-20 at 15 01 59](https://github.com/rectorphp/rector-src/assets/459648/5aa11d1a-3390-41cc-9a68-0e779e068c82)

Reply:
> I think we need more tests on [`main`/tests/Skipper/Skipper/SkipperTest.php](https://github.com/rectorphp/rector-src/blob/main/tests/Skipper/Skipper/SkipperTest.php?rgh-link-date=2024-04-20T08%3A07%3A44Z) to avoid merge and revert

I can't provide more tests, because `*` is not supported on windows and having such tests in your git repo would make it pretty hard to work with the rector-src repo on windows for contributors.